pkg/network/ebpf/c/tracer/events.h

```diff
@@ -124,25 +132,24 @@ static __always_inline void cleanup_conn(void *ctx, conn_tuple_t *tup, struct so
     }
 }
 
+
+// This function is used to flush the conn_close_batch to the perf or ring buffer.
 static __always_inline void flush_conn_close_if_full(void *ctx) {
     u32 cpu = bpf_get_smp_processor_id();
     batch_t *batch_ptr = bpf_map_lookup_elem(&conn_close_batch, &cpu);
-    if (!batch_ptr) {
+    if (!batch_ptr || batch_ptr->len != CONN_CLOSED_BATCH_SIZE) {
         return;
     }
 
-    if (batch_ptr->len == CONN_CLOSED_BATCH_SIZE) {
-        // Here we copy the batch data to a variable allocated in the eBPF stack
-        // This is necessary for older Kernel versions only (we validated this behavior on 4.4.0),
-        // since you can't directly write a map entry to the perf buffer.
-        batch_t batch_copy = {};
-        bpf_memcpy(&batch_copy, batch_ptr, sizeof(batch_copy));
-        batch_ptr->len = 0;
-        batch_ptr->id++;
-
-        // we cannot use the telemetry macro here because of stack size constraints
-        bpf_perf_event_output(ctx, &conn_close_event, cpu, &batch_copy, sizeof(batch_copy));
-    }
+    // Here we copy the batch data to a variable allocated in the eBPF stack
+    // This is necessary for older Kernel versions only (we validated this behavior on 4.4.0),
+    // since you can't directly write a map entry to the perf buffer.
+    batch_t batch_copy = {};
+    bpf_memcpy(&batch_copy, batch_ptr, sizeof(batch_copy));
```

Comment:
right so if I try to change the code to this:
```
    if (ringbuffers_enabled > 0) {
        bpf_ringbuf_output(&conn_close_event, &batch_ptr, sizeof(batch_t), 0);
    } else {
        // Here we copy the batch data to a variable allocated in the eBPF stack
        // This is necessary for older Kernel versions only (we validated this behavior on 4.4.0),
        // since you can't directly write a map entry to the perf buffer.
        batch_t batch_copy = {};
        bpf_memcpy(&batch_copy, batch_ptr, sizeof(batch_copy));
        bpf_perf_event_output(ctx, &conn_close_event, cpu, &batch_copy, sizeof(batch_t));
    }
    batch_ptr->len = 0;
    batch_ptr->id++;
```
we go over the 512 stack limit

Reply:
perhaps we could bring back some of the branching logic I tried in this PR to a new one that also 
- removes this copy
- increases the closed connection batch size

Reply:
Yes, I think we can take this on separately.